### PR TITLE
Development

### PR DESCRIPTION
**Pull request by cornets28**

### CODE REVIEW

**Comment by maelfosso** [comment]
# Under review by Mael FOSSO

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    def my_select #my_select method begins
+        result_array = []
+        
+        self.my_each do |i| #iteration begins
+          yield(i) ? result_array.push(i) : ""
+         end #iteration ends
+         result_array
+      end #my_select method ends
+    
+    def my_all #my_all method begins
```
Comment (maelfosso) [inline enumerable.rb]: What happens if no block given?

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      result_array = []
+      status = false
+      self.my_each do |x| #iteration begins
+        yield(x) ? result_array .push(true) : result_array.push(false)
+      end #iteration ends
+      result_array.my_each do |i| #iteration begins
+        i ? (return status = true) : (return status)
+        end #iteration ends
+    end #my_all method ends
```
Comment (maelfosso) [inline enumerable.rb:38]: It's not working as expected 
```
>> b = [4,3,6,8, 0]
=> [4, 3, 6, 8, 0]
>> b
=> [4, 3, 6, 8, 0]
>> b.all? { |d| d % 2 == 0 }
=> false
>> b.my_all { |d| d % 2 == 0 }
=> true
>> 
```

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    def my_any #my_any method begins
```
Comment (maelfosso) [inline enumerable.rb:40]: Should work if no block given

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+          end
+        end
+        status
+    end #my_any method ends
+
+    def my_none? #my_none method begins
+		((self.my_any { |i| yield(i) == true }) == true) ? false : true 
+    end # my_none method ends
+    
+    def my_inject #my_inject method begins
```
Comment (maelfosso) [inline enumerable.rb]: No working as expected
```
>> [2,4,5].inject { |r, n| r * n }
=> 40
>> [2,4,5].inject(1) { |r, n| r * n }
=> 40
>> [2,4,5].my_inject(1) { |r, n| r * n }
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        5: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):19
        1: from /home/maelfosso/Documents/Microverse/TSE/Ruby/Ruby-Enumerable-development/enumerable.rb:56:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

**Comment by maelfosso** [comment]
Hello,

You did a really nice job. But before moving forward, there are some changes you should do.

- [ ] Please, check the comments above

- [ ] Could you indent your code? It's really hard to read.

Kindly take the comments in consideration and after applying changes, submit a new code review request.

**Comment by AdnanAfsari** [comment]
# **Adnan Afsari working on it**

**Review by AdnanAfsari — APPROVED** [review]
Hello @CornetS28 @SamuelC28 👍 

Congratulation, looks good now i think you have made all the changes previous reviewer requested :)

Please go ahead and merge the branch.

Happy Coding

**Pull request merged**